Assets/MixedRealityToolkit-Examples/Demos/StandardShader/Scripts/MaterialMatrix.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Examples.Demos.StandardShader
+{
+    /// <summary>
+    /// Builds a matrix of spheres demonstrating a spectrum of two material properties.
+    /// </summary>
+    public class MaterialMatrix : MonoBehaviour
+    {
+        [SerializeField]
+        private Material material;
+        [SerializeField]
+        [Range(2, 100)]
+        private int dimension = 5;
+        [SerializeField]
+        [Range(0.0f, 10.0f)]
+        private float positionOffset = 0.1f;
+        [SerializeField]
+        private string firstPropertyName = "_Metallic";
+        [SerializeField]
+        private string secondPropertyName = "_Glossiness";
+
+        private void Awake()
+        {
+            BuildMatrix();
+        }
+
+        public void BuildMatrix()
+        {
+            List<Transform> children = transform.Cast<Transform>().ToList();
+
+            foreach (Transform child in children)
```

Comment:
foreach -> for

Reply:
Thank you for the offer, I wish I would have seen this earlier. But, I think I've just made all the requested changes.